addons/rook_secret_handler.go

```diff
@@ -159,14 +165,19 @@ func getStorageClusterFromRookSecret(ctx context.Context, client client.Client,
 			var cephCluster rookv1.CephCluster
 			err := client.Get(ctx, types.NamespacedName{Namespace: secret.Namespace, Name: v.Name}, &cephCluster)
 			if err != nil {
-				return "", fmt.Errorf("unable to fetch ceph cluster err: %v", err)
+				return "", fmt.Errorf("failed to fetch CephCluster '%s' referenced by secret '%s' in namespace '%s': %v", v.Name, secret.Name, secret.Namespace, err)
 			}
 
+			found := false
 			for _, owner := range cephCluster.ObjectMeta.OwnerReferences {
-				if owner.Kind != "StorageCluster" {
-					continue
+				if owner.Kind == "StorageCluster" {
+					storageCluster = owner.Name
+					found = true
+					break
```

Comment:
This is unrelated to logging changes. Keep it in a separate PR.

Reply:
Can we keep this as a separate commit ?

Reply:
Couple of small refactors came to light during changing the log messages which is why i added them